Sources/Apollo/HTTPNetworkTransport.swift

```diff
@@ -41,11 +41,24 @@ public struct GraphQLHTTPResponseError: Error, LocalizedError {
   }
 }
 
+/// Used to intercept the http requests/errors/responses. This allows modifying url requests dynamically without recreating transports
+public protocol HTTPNetworkTransportDelegate: class {
+  func intercept() -> HTTPNetworkTransportInterception
+}
+
+/// An encapsulation of state for a single request (response/error) pair. If any method returns non-null it will be used in the http pipeline
+public protocol HTTPNetworkTransportInterception {
+  func willSend(request: URLRequest) -> URLRequest?
```

Comment:
When using willSend I can access request fields but I cannot add headers (request is immutable object)